bug report

---------- Bug report ----------

Please include a description about what actions you were performing when the exception occurred:
(You can edit text directly in this window)

1.
2.
3.

If possible, please send us the .ork file that caused the bug.

Include your email address (optional; it helps if we can contact you in case we need additional information):




(Do not modify anything below this line.)
---------- Exception stack trace ----------
java.lang.RuntimeException: java.io.FileNotFoundException: C:\Program Files\OpenRocket\ShellFolder: 0x11\nose cone (1).obj (The filename, directory name, or volume label syntax is incorrect) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:233) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220) at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1718) at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1705) at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1698) at info.openrocket.swing.gui.main.RocketActions$ExportOBJAction.actionPerformed(RocketActions.java:1274) at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072) at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297) at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626) at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389) at java.desktop/java.awt.Component.processEvent(Component.java:6391) at java.desktop/java.awt.Container.processEvent(Container.java:2266) at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744) at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) Caused by: java.io.FileNotFoundException: C:\Program Files\OpenRocket\ShellFolder: 0x11\nose cone (1).obj (The filename, directory name, or volume label syntax is incorrect) at java.base/java.io.FileOutputStream.open0(Native Method) at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230) ... 43 more

---------- Thread information ----------
Thread[AWT-EventQueue-0,6,main]

---------- System information ----------
OpenRocket version: 24.12
OpenRocket source: default
OpenRocket location: C:\Program Files\OpenRocket\.\jar\OpenRocket-24.12.jar
User-defined thrust curves location: C:\Users\maval\AppData\Roaming\OpenRocket\ThrustCurves
LAF: com.formdev.flatlaf.FlatLightLaf
JOGL version: null
Current default locale: en_US
JVM Arguments: -Dexe4j.semaphoreName=Local\c:_program_files_openrocket_openrocket.exe2 -Dexe4j.isInstall4j=true -Dexe4j.moduleName=C:\Program Files\OpenRocket\OpenRocket.exe -Dexe4j.tempDir= -Dexe4j.unextractedPosition=0 -Djava.library.path=c:\program files\openrocket\jre\bin;C:\Program Files (x86)\Microsoft\Edge\Application;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Windows\system32;C:\Program Files\dotnet\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Users\maval\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Processing\ -Dexe4j.consoleCodepage=cp0 -Dapple.awt.application.appearance=system -Dsun.java2d.noddraw=true -Dsun.java2d.d3d=false -Dsun.java2d.ddforcevram=true -Dsun.java2d.ddblit=false -Dswing.useflipBufferStrategy=True --add-exports=java.base/java.lang=ALL-UNNAMED --add-exports=java.desktop/sun.awt=ALL-UNNAMED --add-exports=java.desktop/sun.java2d=ALL-UNNAMED -Dinstall4j.launcherId=59 -Dinstall4j.swt=false
System properties:
apple.awt.application.appearance=system
exe4j.consoleCodepage=cp0
exe4j.isInstall4j=true
exe4j.launchName=C:\Program Files\OpenRocket\OpenRocket.exe
exe4j.moduleName=C:\Program Files\OpenRocket\OpenRocket.exe
exe4j.semaphoreName=Local\c:_program_files_openrocket_openrocket.exe2
exe4j.tempDir=
exe4j.unextractedPosition=0
file.encoding=Cp1252
file.separator=\
flatlaf.uiScale=1.0
install4j.appDir=C:\Program Files\OpenRocket\
install4j.exeDir=C:\Program Files\OpenRocket\
install4j.launcherId=59
install4j.swt=false
java.class.path=C:\Program Files\OpenRocket\.install4j\i4jruntime.jar;C:\Program Files\OpenRocket\.\jar\OpenRocket-24.12.jar
java.class.version=61.0
java.home=c:\program files\openrocket\jre
java.io.tmpdir=C:\Users\maval\AppData\Local\Temp\
java.library.path=c:\program files\openrocket\jre\bin;C:\Program Files (x86)\Microsoft\Edge\Application;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Windows\system32;C:\Program Files\dotnet\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Users\maval\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Processing\
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=17.0.16+12-LTS
java.specification.maintenance.version=1
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=17
java.util.Arrays.useLegacyMergeSort=true
java.vendor=BellSoft
java.vendor.url=https://bell-sw.com/
java.vendor.url.bug=https://bell-sw.com/support
java.version=17.0.16
java.version.date=2025-07-15
java.vm.compressedOopsMode=Zero based
java.vm.info=mixed mode
java.vm.name=OpenJDK 64-Bit Server VM
java.vm.specification.name=Java Virtual Machine Specification
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=17
java.vm.vendor=BellSoft
java.vm.version=17.0.16+12-LTS
jdk.debug=release
jnlp.jogamp.tmp.cache.root=jln14120806401960181108
line.separator=\u000d\u000a
native.encoding=Cp1252
os.arch=amd64
os.name=Windows 11
os.version=10.0
path.separator=;
sun.arch.data.model=64
sun.awt.enableExtraMouseButtons=true
sun.awt.exception.handler=info.openrocket.swing.gui.main.SwingExceptionHandler$AwtHandler
sun.boot.library.path=c:\program files\openrocket\jre\bin
sun.cpu.endian=little
sun.cpu.isalist=amd64
sun.io.unicode.encoding=UnicodeLittle
sun.java.command=C:\Program Files\OpenRocket\OpenRocket.exe
sun.java2d.d3d=false
sun.java2d.ddblit=false
sun.java2d.ddforcevram=true
sun.java2d.noddraw=true
sun.jnu.encoding=Cp1252
sun.management.compiler=HotSpot 64-Bit Tiered Compilers
sun.os.patch.level=
swing.useflipBufferStrategy=True
user.country=US
user.dir=C:\Program Files\OpenRocket
user.home=C:\Users\maval
user.language=en
user.name=maval
user.script=
user.timezone=America/New_York
user.variant=
---------- Error log ----------
1 0.000 INFO (SwingStartup.java:75) Starting up OpenRocket version 24.12
2 0.000 INFO (SwingStartup.java:84) Checking for graphics head
3 0.000 INFO (SwingStartup.java:95) Moving startup to EDT
4 0.032 INFO (SwingStartup.java:187) Initializing the splash screen
5 0.047 INFO (SwingStartup.java:191) Registering exception handler
6 0.063 INFO (SwingStartup.java:197) Loading databases
7 2.668 INFO (SwingStartup.java:201) Using default locale en_US
8 2.712 INFO (SwingStartup.java:201) Set up translation for locale en_US, debug.currentFile=messages.properties
9 2.752 INFO (TraceException.java:26) Loading component presets from datafiles/components
10 2.752 INFO (TraceException.java:26) Starting reading serialized motor database
11 2.772 INFO (SwingStartup.java:280) Starting update check
12 2.777 INFO (SwingStartup.java:210) Setting LAF
18 3.347 INFO (TraceException.java:26) Ending reading serialized motor database, motorCount=1936
19 3.358 INFO (TraceException.java:26) Starting reading user-defined motors
20 3.364 INFO (TraceException.java:26) Ending reading user-defined motors, motorCount=1936
22 4.001 INFO (UITheme.java:1955) Setting UI scale factor to 1.0
23 4.001 INFO (UITheme.java:1959) Loading custom fonts
26 4.075 INFO (UITheme.java:1966) Setting global font to 14 Inter-Regular 0.0
31 4.228 INFO (TraceException.java:26) Current build is latest version
34 4.278 INFO (SwingStartup.java:234) Opening main application window
45 4.350 INFO (BasicFrame.java:1354) Starting OpenFileWorker
46 4.353 INFO (SwingWorkerDialog.java:133) Running SwingWorker info.openrocket.swing.gui.util.OpenFileWorker@30a85098
47 4.368 INFO (OpenFileWorker.java:73) Loading .ork file
80 4.572 INFO (OpenFileWorker.java:73) Loading done
87 4.584 INFO (SwingWorkerDialog.java:151) Worker completed before opening dialog
138 5.017 INFO (TraceException.java:26) Starting reading user-defined component presets
139 5.019 INFO (TraceException.java:26) Ending reading user-defined component presets, presetCount=0
195 18.992 INFO (SwingStartup.java:242) Checking update status
196 18.999 INFO (SwingStartup.java:103) Startup complete
268 117.388 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Sustainer
272 117.465 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Body Tube/afd25ee7 / A8 - Armed @: 0.0
279 117.467 INFO (SimulationWorker.java:55) Igniting motor: Body Tube/afd25ee7 / A8 - Armed @0.0
4982 118.021 INFO (SimulationWorker.java:55) <<Finished simulating branch: Sustainer curTime:8.834267701079021 finTime:8.834267701079021
4983 118.021 INFO (SimulationWorker.java:55) Warnings at the end of simulation: Messages[No recovery device defined in the simulation.]
5122 148.971 USER (BasicFrame.java:504) Print action selected
5123 517.965 INFO (PaperSize.java:77) Selecting default paper size based on user.country: Letter (ANSI A)
5124 518.147 INFO (UITheme.java:1955) Setting UI scale factor to 1.0
5125 518.147 INFO (UITheme.java:1959) Loading custom fonts
5128 518.183 INFO (UITheme.java:1966) Setting global font to 14 Inter-Regular 0.0
5171 518.689 INFO (UITheme.java:1955) Setting UI scale factor to 1.0
5172 518.689 INFO (UITheme.java:1959) Loading custom fonts
5175 518.724 INFO (UITheme.java:1966) Setting global font to 14 Inter-Regular 0.0
5285 650.058 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Body Tube' to info.openrocket.core.document.OpenRocketDocument@494d4a6e, document was in clean state
5630 1591.818 ERROR (SwingExceptionHandler.java:49) Handling uncaught exception on thread=Thread[AWT-EventQueue-0,6,main]
> java.lang.RuntimeException: java.io.FileNotFoundException: C:\Program Files\OpenRocket\ShellFolder: 0x11\rockey 1.obj (The filename, directory name, or volume label syntax is incorrect) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:233) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220) > at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1718) > at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1705) > at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1698) > at info.openrocket.swing.gui.main.BasicFrame$8.actionPerformed(BasicFrame.java:459) > at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) > at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) > at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) > at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) > at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072) > at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626) > at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389) > at java.desktop/java.awt.Component.processEvent(Component.java:6391) > at java.desktop/java.awt.Container.processEvent(Container.java:2266) > at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) > at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) > at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) > at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744) > at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) > at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) > Caused by: java.io.FileNotFoundException: C:\Program Files\OpenRocket\ShellFolder: 0x11\rockey 1.obj (The filename, directory name, or volume label syntax is incorrect) > at java.base/java.io.FileOutputStream.open0(Native Method) > at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230) > ... 42 more
5682 1591.824 INFO (SwingExceptionHandler.java:62) Exception handler running on EDT, showing dialog
5683 1591.824 INFO (SwingExceptionHandler.java:202) Showing Exception dialog
5684 1597.577 USER (SwingExceptionHandler.java:221) User requested sending bug report
5831 1613.360 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Nose Cone' to info.openrocket.core.document.OpenRocketDocument@494d4a6e, document was in clean state
6323 1623.477 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Body Tube' to info.openrocket.core.document.OpenRocketDocument@494d4a6e, document was in clean state
6813 1630.924 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Trapezoidal Fin Set' to info.openrocket.core.document.OpenRocketDocument@494d4a6e, document was in clean state
7111 0.000 WARN (-) ===== 309 WARN lines removed =====
7111 1634.884 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@5fd0c293 but it was not registered
7155 1634.888 WARN (BooleanModel.java:352) Invalidating BooleanModel[internal value] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@2020b9af]
7160 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@4d66a1da while still having listeners [javax.swing.JSpinner$ModelListener@72d68220]
7165 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@75a41281 while still having listeners [javax.swing.JSpinner$ModelListener@33254997]
7170 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@6b1a4167 while still having listeners [javax.swing.JSpinner$ModelListener@72eac462, javax.swing.JSlider$ModelListener@727d31f0, javax.swing.plaf.basic.BasicSliderUI$Handler@3726d5e8]
7175 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@7c40ce82 while still having listeners [javax.swing.JSpinner$ModelListener@1a27e143]
7180 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@700dd922 while still having listeners [javax.swing.JSpinner$ModelListener@55851399]
7185 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@3e30ac54 while still having listeners [javax.swing.JSpinner$ModelListener@435469d0, javax.swing.JSlider$ModelListener@2f0cbe22, javax.swing.plaf.basic.BasicSliderUI$Handler@5a3e4954]
7190 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@3aaa5948 while still having listeners [javax.swing.JSpinner$ModelListener@54735ae6, javax.swing.JSlider$ModelListener@70cdadc5, javax.swing.plaf.basic.BasicSliderUI$Handler@3ea76d50]
7199 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@c85f7ef while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@26df69d1]
7204 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@e5cca7f while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@53d7715e]
7209 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@3905ccc7 while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@6cda28c6]
7214 1634.888 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@23a8c138 while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@53d7715e]
7219 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@33851d0e but it was not registered
7223 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@42c96b5f but it was not registered
7227 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7200e885 but it was not registered
7231 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4895037d but it was not registered
7235 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@70464b0f but it was not registered
7239 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@374d7e7b but it was not registered
7243 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@25374a15 but it was not registered
7247 1634.888 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@2d9b0427 but it was not registered
7251 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@78d5e093 but it was not registered
7255 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1e3b72a9 but it was not registered
7259 1634.890 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@6ba4231c while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1118f23c]
7264 1634.890 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1be1cb2f while still having listeners [info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@1118f23c]
7266 0.000 DEBUG (-) ===== 920 DEBUG lines removed =====
7266 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[TrapezoidFinSet:RootChord]
7267 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@1be1cb2f
7269 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4aa9a79f but it was not registered
7270 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[TrapezoidFinSet:SweepAngle]
7271 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@55a8c224
7273 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@792165e1 but it was not registered
7274 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.rocketcomponent.TrapezoidFinSet:CrossSection]
7275 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@46737b23
7277 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@89adeda but it was not registered
7278 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[TrapezoidFinSet:Thickness]
7279 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@7940f21b
7281 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@348a80e2 but it was not registered
7282 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.rocketcomponent.TrapezoidFinSet:AxialMethod]
7283 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@4cf8fce
7285 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4cef9097 but it was not registered
7286 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[TrapezoidFinSet:AxialOffset]
7287 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@b0451b2
7289 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@733e83f2 but it was not registered
7290 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[TrapezoidFinSet:BaseRotation]
7291 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@4c23ad6e
7293 1634.890 WARN (ModelInvalidator.java:111) Attempted to remove ComponentChangeListener info.openrocket.swing.gui.adaptors.MaterialModel@6a3533b2 but it was not registered
7294 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: info.openrocket.swing.gui.adaptors.MaterialModel@6a3533b2
7295 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@57ff9b91
7297 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7e974250 but it was not registered
7298 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.rocketcomponent.TrapezoidFinSet:Finish]
7299 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@4fdbd7db
7301 1634.890 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7f6aa294 but it was not registered
7302 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[TrapezoidFinSet:FilletRadius]
7303 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@57501195
7305 1634.890 WARN (ModelInvalidator.java:111) Attempted to remove ComponentChangeListener info.openrocket.swing.gui.adaptors.MaterialModel@49ec01b3 but it was not registered
7306 1634.890 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: info.openrocket.swing.gui.adaptors.MaterialModel@49ec01b3
7307 1634.890 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@197fa3b3
7309 2844.245 DEBUG (RocketFigure3d.java:573) 3D Figure Updated
7310 2844.261 DEBUG (TextureCache.java:51) ClearCaches
7311 0.000 VBOSE (-) ===== 5921 VBOSE lines removed =====
7311 2845.785 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.736476 user scale, 3479.41 overall scale
7312 2914.634 USER (BasicFrame.java:528) Properties selected
7313 2914.636 VBOSE (ComponentConfigDialog.java:106) Added ComponentChangeListener info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog7,0,0,0x0,invalid,hidden,layout=java.awt.BorderLayout,MODELESS,title=,defaultCloseOperation=HIDE_ON_CLOSE,rootPane=javax.swing.JRootPane[,0,0,0x0,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true], current number of listeners is 30
7314 2914.636 VBOSE (BooleanModel.java:111) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@3194f4c0, current number of listeners is 31
7315 2914.636 WARN (BooleanModel.java:151) Attempted to add ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@3194f4c0 but it was already registered
7316 2914.636 VBOSE (BooleanModel.java:111) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4718618d, current number of listeners is 32
7317 2914.636 WARN (BooleanModel.java:151) Attempted to add ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4718618d but it was already registered
7318 2914.636 VBOSE (DoubleModel.java:732) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7e781e63, current number of listeners is 33
7319 2914.636 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@7ae80c28 adding listener (total 1): javax.swing.JSpinner$ModelListener@40feae64
7320 2914.636 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@4ef5be8c adding listener (total 1): info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@7b4da4b8
7321 2914.636 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@7ae80c28 adding listener (total 2): javax.swing.JSlider$ModelListener@2ee1fb23
7322 2914.636 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@7ae80c28 adding listener (total 3): javax.swing.plaf.basic.BasicSliderUI$Handler@69cb767a
7323 2914.636 VBOSE (BooleanModel.java:111) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1c57a998, current number of listeners is 34
7324 2914.636 WARN (BooleanModel.java:151) Attempted to add ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1c57a998 but it was already registered
7325 2914.636 VBOSE (BooleanModel.java:111) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@f04b8b7, current number of listeners is 35
7326 2914.636 WARN (BooleanModel.java:151) Attempted to add ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@f04b8b7 but it was already registered
7327 2914.640 VBOSE (DoubleModel.java:732) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@27fe4b8f, current number of listeners is 36
7328 2914.640 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@d159ec7 adding listener (total 1): javax.swing.JSpinner$ModelListener@11207a43
7329 2914.643 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@bb79de1 adding listener (total 1): info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@19571cb6
7330 2914.643 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@7424c24d adding listener (total 1): info.openrocket.swing.gui.adaptors.DoubleModel$ValueSliderModel@19571cb6
7331 2914.643 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@d159ec7 adding listener (total 2): javax.swing.JSlider$ModelListener@703b6aba
7332 2914.643 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@d159ec7 adding listener (total 3): javax.swing.plaf.basic.BasicSliderUI$Handler@4754a9c9
7333 2914.643 VBOSE (BooleanModel.java:111) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@34a32f06, current number of listeners is 37
7334 2914.643 WARN (BooleanModel.java:151) Attempted to add ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@34a32f06 but it was already registered
7335 2914.643 VBOSE (BooleanModel.java:111) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@6c164654, current number of listeners is 38
7336 2914.643 WARN (BooleanModel.java:151) Attempted to add ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@6c164654 but it was already registered
7337 2914.643 VBOSE (DoubleModel.java:732) Added ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@5f9acb5f, current number of listeners is 39
7338 2914.645 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@160021e adding listener (total 1): javax.swing.JSpinner$ModelListener@4c70b73
7339 2914.647 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@160021e adding listener (total 2): javax.swing.JSlider$ModelListener@20dd3ed6
7340 2914.647 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@160021e adding listener (total 3): javax.swing.plaf.basic.BasicSliderUI$Handler@212ca821
7341 2914.723 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Rocket' to info.openrocket.core.document.OpenRocketDocument@494d4a6e, document was in clean state
7342 2916.791 DEBUG (GUIUtil.java:312) Clearing all models of window info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog7,386,267,551x519,invalid,hidden,layout=java.awt.BorderLayout,MODELESS,title=Rocket configuration,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,537x511,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
7343 2916.791 DEBUG (GUIUtil.java:314) Adding object into collectable list: info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog7,386,267,551x519,invalid,hidden,layout=java.awt.BorderLayout,MODELESS,title=Rocket configuration,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,537x511,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
7344 2916.791 WARN (BooleanModel.java:352) Invalidating BooleanModel[Rocket:MassOverridden] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@15388a0b]
7345 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@738aa6bd
7346 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@3194f4c0, current number of listeners is 38
7347 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[Rocket:MassOverridden]
7348 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@738aa6bd
7349 2916.791 WARN (BooleanModel.java:352) Invalidating BooleanModel[Rocket:SubcomponentsOverriddenMass] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@4c18c4b4]
7350 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@2f9b33e4
7351 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@4718618d, current number of listeners is 37
7352 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[Rocket:SubcomponentsOverriddenMass]
7353 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@2f9b33e4
7354 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@7ae80c28
7355 2916.791 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@7ae80c28 while still having listeners [javax.swing.JSpinner$ModelListener@40feae64, javax.swing.JSlider$ModelListener@2ee1fb23, javax.swing.plaf.basic.BasicSliderUI$Handler@69cb767a]
7356 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@7e781e63, current number of listeners is 36
7357 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[Rocket:OverrideMass]
7358 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@7ae80c28
7359 2916.791 WARN (BooleanModel.java:352) Invalidating BooleanModel[Rocket:CGOverridden] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@7e229cce]
7360 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@58bb62f9
7361 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1c57a998, current number of listeners is 35
7362 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[Rocket:CGOverridden]
7363 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@58bb62f9
7364 2916.791 WARN (BooleanModel.java:352) Invalidating BooleanModel[Rocket:SubcomponentsOverriddenCG] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@520c303e]
7365 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@247e3d88
7366 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@f04b8b7, current number of listeners is 34
7367 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[Rocket:SubcomponentsOverriddenCG]
7368 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@247e3d88
7369 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@d159ec7
7370 2916.791 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@d159ec7 while still having listeners [javax.swing.JSpinner$ModelListener@11207a43, javax.swing.JSlider$ModelListener@703b6aba, javax.swing.plaf.basic.BasicSliderUI$Handler@4754a9c9]
7371 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@27fe4b8f, current number of listeners is 33
7372 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[Rocket:OverrideCGX]
7373 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@d159ec7
7374 2916.791 WARN (BooleanModel.java:352) Invalidating BooleanModel[Rocket:CDOverridden] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@4ed00125]
7375 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@4dff1eac
7376 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@34a32f06, current number of listeners is 32
7377 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[Rocket:CDOverridden]
7378 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@4dff1eac
7379 2916.791 WARN (BooleanModel.java:352) Invalidating BooleanModel[Rocket:SubcomponentsOverriddenCD] while still having listeners [javax.swing.AbstractButton$ButtonActionPropertyChangeListener@7ec15aed]
7380 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@2b06aac0
7381 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@6c164654, current number of listeners is 31
7382 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: BooleanModel[Rocket:SubcomponentsOverriddenCD]
7383 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@2b06aac0
7384 2916.791 VBOSE (ModelInvalidator.java:101) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@160021e
7385 2916.791 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@160021e while still having listeners [javax.swing.JSpinner$ModelListener@4c70b73, javax.swing.JSlider$ModelListener@20dd3ed6, javax.swing.plaf.basic.BasicSliderUI$Handler@212ca821]
7386 2916.791 VBOSE (ModelInvalidator.java:109) Removed ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@5f9acb5f, current number of listeners is 30
7387 2916.791 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[Rocket:OverrideCD]
7388 2916.791 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@160021e
7389 2916.791 VBOSE (ComponentConfigDialog.java:76) Removed ComponentChangeListener info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog7,386,267,551x519,invalid,hidden,layout=java.awt.BorderLayout,MODELESS,title=Rocket configuration,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,537x511,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true], current number of listeners is 29
7390 2984.271 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.736476 user scale, 3479.41 overall scale
7391 2984.287 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.736476 user scale, 3479.41 overall scale
7392 2987.761 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.736476 user scale, 3479.41 overall scale
7393 3013.680 DEBUG (RocketFigure3d.java:573) 3D Figure Updated
7394 3013.680 DEBUG (TextureCache.java:51) ClearCaches
7395 3017.624 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.736476 user scale, 3479.41 overall scale
7396 3017.631 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.736476 user scale, 3479.41 overall scale
7397 3024.004 DEBUG (FileHelper.java:149) File name does not contain extension, adding 'obj'
7398 3024.006 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@15303c4b adding listener (total 1): javax.swing.JSpinner$ModelListener@1b62e472
7399 3024.007 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@15303c4b adding listener (total 2): info.openrocket.swing.gui.choosers.OBJOptionChooser$10@762f8dc9
7400 3024.007 VBOSE (ModelInvalidator.java:55) info.openrocket.swing.gui.adaptors.ModelInvalidator@15303c4b adding listener (total 3): info.openrocket.swing.gui.choosers.OBJOptionChooser$7@5f89ca4
7401 3024.009 DEBUG (DoubleModel.java:873) Setting value 1000.0 for DoubleModel[OBJExportOptions:ScalingDouble]
7402 3035.378 DEBUG (FileHelper.java:149) File name does not contain extension, adding 'obj'
7403 3035.530 ERROR (SwingExceptionHandler.java:49) Handling uncaught exception on thread=Thread[AWT-EventQueue-0,6,main]
> java.lang.RuntimeException: java.io.FileNotFoundException: C:\Program Files\OpenRocket\ShellFolder: 0x11\nose cone (1).obj (The filename, directory name, or volume label syntax is incorrect) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:233) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220) > at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1718) > at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1705) > at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1698) > at info.openrocket.swing.gui.main.RocketActions$ExportOBJAction.actionPerformed(RocketActions.java:1274) > at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) > at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) > at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) > at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) > at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028) > at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072) > at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297) > at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626) > at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389) > at java.desktop/java.awt.Component.processEvent(Component.java:6391) > at java.desktop/java.awt.Container.processEvent(Container.java:2266) > at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) > at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) > at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) > at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) > at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780) > at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) > at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) > at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747) > at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745) > at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) > at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) > at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744) > at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) > at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) > at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) > at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) > Caused by: java.io.FileNotFoundException: C:\Program Files\OpenRocket\ShellFolder: 0x11\nose cone (1).obj (The filename, directory name, or volume label syntax is incorrect) > at java.base/java.io.FileOutputStream.open0(Native Method) > at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235) > at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155) > at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230) > ... 43 more
7406 0.000 STDERR (-) ===== 53 STDERR lines removed =====
7406 3035.530 STDERR (SwingExceptionHandler.java:50) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.doExport(OBJExporterFactory.java:220)
7407 3035.530 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.BasicFrame.saveWavefrontOBJFile(BasicFrame.java:1718)
7408 3035.530 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.BasicFrame.saveAsWavefrontOBJ(BasicFrame.java:1705)
7409 3035.530 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.BasicFrame.exportWavefrontOBJAction(BasicFrame.java:1698)
7410 3035.530 STDERR (SwingExceptionHandler.java:50) at info.openrocket.swing.gui.main.RocketActions$ExportOBJAction.actionPerformed(RocketActions.java:1274)
7411 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972)
7412 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313)
7413 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
7414 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
7415 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:374)
7416 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028)
7417 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072)
7418 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
7419 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626)
7420 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389)
7421 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.processEvent(Component.java:6391)
7422 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Container.processEvent(Container.java:2266)
7423 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
7424 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
7425 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
7426 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
7427 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
7428 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
7429 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
7430 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
7431 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
7432 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775)
7433 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
7434 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
7435 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
7436 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
7437 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
7438 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747)
7439 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
7440 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
7441 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
7442 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744)
7443 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
7444 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
7445 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
7446 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
7447 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
7448 3035.530 STDERR (SwingExceptionHandler.java:50) at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
7449 3035.530 STDERR (SwingExceptionHandler.java:50) Caused by: java.io.FileNotFoundException: C:\Program Files\OpenRocket\ShellFolder: 0x11\nose cone (1).obj (The filename, directory name, or volume label syntax is incorrect)
7450 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.open0(Native Method)
7451 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293)
7452 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235)
7453 3035.530 STDERR (SwingExceptionHandler.java:50) at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:155)
7454 3035.530 STDERR (SwingExceptionHandler.java:50) at info.openrocket.core.file.wavefrontobj.export.OBJExporterFactory.writeObj(OBJExporterFactory.java:230)
7455 3035.530 STDERR (SwingExceptionHandler.java:50) ... 43 more
7456 3035.530 INFO (SwingExceptionHandler.java:62) Exception handler running on EDT, showing dialog
7457 3035.530 INFO (SwingExceptionHandler.java:202) Showing Exception dialog
7458 3050.818 USER (SwingExceptionHandler.java:221) User requested sending bug report
7459 3050.818 DEBUG (BugReportDialog.java:230) Found jar file using codeSource
---------- End of bug report ----------

--- a/core/src/main/resources/l10n/messages.properties
+++ b/core/src/main/resources/l10n/messages.properties
@@ -1969,6 +1969,8 @@ RASAeroExport.error33 = Invalid component '%s' in sustainer stage.
 ! SaveAsFileChooser
 SaveAsFileChooser.illegalFilename.title = Illegal filename
 SaveAsFileChooser.illegalFilename.message = The filename '%s' may not contain the character ' %c '. Please remove it.
+SaveAsFileChooser.invalidDirectory.title = Invalid save location
+SaveAsFileChooser.invalidDirectory.message = Select a folder on a local or network drive before saving. Files cannot be saved directly in this location.
 
 ! StorageOptionChooser
 StorageOptChooser.lbl.Simdatatostore = Simulated data to store:

--- a/swing/src/main/java/info/openrocket/swing/gui/widgets/SaveFileChooser.java
+++ b/swing/src/main/java/info/openrocket/swing/gui/widgets/SaveFileChooser.java
@@ -9,6 +9,7 @@ import javax.swing.ActionMap;
 import javax.swing.JFileChooser;
 import javax.swing.JOptionPane;
 import javax.swing.SwingUtilities;
+import javax.swing.filechooser.FileSystemView;
 import java.awt.event.ActionEvent;
 import java.io.File;
 import java.util.List;
@@ -115,6 +116,12 @@ public class SaveFileChooser extends JFileChooser {
 
     @Override
     public void approveSelection() {
+        File targetDirectory = getFileSelectionMode() == JFileChooser.DIRECTORIES_ONLY ? currentFile : cwd;
+        if (!isFileSystemDirectory(targetDirectory, getFileSystemView())) {
+            showInvalidSaveLocationWarning();
+            return;
+        }
+
         Character c = FileUtils.getIllegalFilenameChar(fileName);
         if (c != null) {
             // Illegal character found
@@ -128,6 +135,30 @@ public class SaveFileChooser extends JFileChooser {
             setCurrentDirectory(cwd);
             super.approveSelection();
         }
+    }
+
+    /**
+     * Shows the warning used when a virtual shell location cannot contain a saved file.
+     */
+    protected void showInvalidSaveLocationWarning() {
+        JOptionPane.showMessageDialog(getParent(),
+                trans.get("SaveAsFileChooser.invalidDirectory.message"),
+                trans.get("SaveAsFileChooser.invalidDirectory.title"),
+                JOptionPane.WARNING_MESSAGE);
+    }
+
+    /**
+     * Checks that a save destination is backed by the filesystem.  Windows exposes
+     * virtual shell nodes such as "This PC" as {@link File} instances even though
+     * files cannot be written directly inside them.
+     *
+     * @param directory the directory selected as, or containing, the save target
+     * @param fileSystemView the platform file-system view used by the chooser
+     * @return {@code true} when the directory is a real filesystem directory
+     */
+    static boolean isFileSystemDirectory(File directory, FileSystemView fileSystemView) {
+        return directory != null && fileSystemView != null
+                && fileSystemView.isFileSystem(directory) && directory.isDirectory();
     }
 
     /**

--- a/swing/src/test/java/info/openrocket/swing/gui/widgets/SaveFileChooserTest.java
+++ b/swing/src/test/java/info/openrocket/swing/gui/widgets/SaveFileChooserTest.java
@@ -1,0 +1,125 @@
+package info.openrocket.swing.gui.widgets;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.swing.JFileChooser;
+import javax.swing.filechooser.FileSystemView;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Tests validation of filesystem-backed save destinations.
+ */
+public class SaveFileChooserTest {
+	@TempDir
+	Path tempDirectory;
+
+	/**
+	 * A normal directory accepted by the platform file-system view is a valid destination.
+	 */
+	@Test
+	public void testAcceptsFileSystemDirectory() {
+		File directory = tempDirectory.toFile();
+		FileSystemView fileSystemView = FileSystemView.getFileSystemView();
+
+		assertTrue(SaveFileChooser.isFileSystemDirectory(directory, fileSystemView));
+	}
+
+	/**
+	 * Virtual Windows shell nodes, such as "This PC", must not be used as save directories.
+	 */
+	@Test
+	public void testRejectsVirtualShellDirectory() {
+		File virtualDirectory = tempDirectory.toFile();
+		FileSystemView fileSystemView = mock(FileSystemView.class);
+		when(fileSystemView.isFileSystem(virtualDirectory)).thenReturn(false);
+
+		assertFalse(SaveFileChooser.isFileSystemDirectory(virtualDirectory, fileSystemView));
+	}
+
+	/**
+	 * A filesystem path must also exist as a directory before it can contain a saved file.
+	 */
+	@Test
+	public void testRejectsMissingDirectory() {
+		File directory = tempDirectory.resolve("missing").toFile();
+		FileSystemView fileSystemView = FileSystemView.getFileSystemView();
+
+		assertFalse(SaveFileChooser.isFileSystemDirectory(directory, fileSystemView));
+	}
+
+	/**
+	 * Approval must remain inside the chooser when the current location is a virtual shell node.
+	 */
+	@Test
+	public void testVirtualShellDirectoryPreventsApproval() {
+		File directory = tempDirectory.toFile();
+		FileSystemView fileSystemView = mock(FileSystemView.class);
+		when(fileSystemView.isFileSystem(directory)).thenReturn(false);
+		TestSaveFileChooser chooser = new TestSaveFileChooser();
+		chooser.setCurrentDirectory(directory);
+		chooser.setSelectedFile(new File(directory, "rocket.obj"));
+		chooser.setTestFileSystemView(fileSystemView);
+		AtomicBoolean approved = new AtomicBoolean(false);
+		chooser.addActionListener(event -> approved.set(
+				JFileChooser.APPROVE_SELECTION.equals(event.getActionCommand())));
+
+		chooser.approveSelection();
+
+		assertTrue(chooser.isInvalidLocationWarningShown());
+		assertFalse(approved.get());
+	}
+
+	/**
+	 * Approval must proceed normally when the current location is a filesystem directory.
+	 */
+	@Test
+	public void testFileSystemDirectoryAllowsApproval() {
+		File directory = tempDirectory.toFile();
+		TestSaveFileChooser chooser = new TestSaveFileChooser();
+		chooser.setCurrentDirectory(directory);
+		chooser.setSelectedFile(new File(directory, "rocket.obj"));
+		AtomicBoolean approved = new AtomicBoolean(false);
+		chooser.addActionListener(event -> approved.set(
+				JFileChooser.APPROVE_SELECTION.equals(event.getActionCommand())));
+
+		chooser.approveSelection();
+
+		assertFalse(chooser.isInvalidLocationWarningShown());
+		assertTrue(approved.get());
+	}
+
+	/**
+	 * Test chooser that records warnings without displaying a modal dialog.
+	 */
+	private static final class TestSaveFileChooser extends SaveFileChooser {
+		private FileSystemView testFileSystemView;
+		private boolean invalidLocationWarningShown;
+
+		@Override
+		public FileSystemView getFileSystemView() {
+			return testFileSystemView != null ? testFileSystemView : super.getFileSystemView();
+		}
+
+		@Override
+		protected void showInvalidSaveLocationWarning() {
+			invalidLocationWarningShown = true;
+		}
+
+		private void setTestFileSystemView(FileSystemView fileSystemView) {
+			testFileSystemView = fileSystemView;
+		}
+
+		private boolean isInvalidLocationWarningShown() {
+			return invalidLocationWarningShown;
+		}
+	}
+}

